commands/hugo.go

```diff
@@ -781,6 +781,13 @@ func (c *commandeer) newWatcher(dirList ...string) (*watcher.Batcher, error) {
 						if ev.Op&fsnotify.Chmod == fsnotify.Chmod {
 							continue
 						}
+						if ev.Op&fsnotify.Remove == fsnotify.Remove {
+							for _, configFile := range c.configFiles {
+								for watcher.Add(configFile) != nil {
```

Comment:
I don't remember this for loop in the original PR. What is the motivation behind this? If it errors once if will continue to error forever, I assume.

Reply:
Thank you again for the code review.

> I don't remember this for loop in the original PR.

I have kept the original commit for this PR is here: https://github.com/anthonyfok/hugo/commit/b73d8107fe8f654f3915f30a14185bdd9417b968

> What is the motivation behind this?

watcher.Add(configFile)  returns non-nil when the configFile does not exist.

This for loop is to ensure the configFile is indeed added to the fsnotify watcher.
On a slow/busy system, it may take longer than 100ms for the removed file to reappear, sometimes over 1 second, so it went through the loop like 10 times until config.toml re-appear.

> If it errors once if will continue to error forever, I assume.

Not in my local testing, when I kept saving config.toml from within Vim or GNOME gedit.

However, I think you correctly point out a potential danger of getting stuck in this loop, as in: what if watcher.Add() fails not because config.toml does not exist, but for some other reasons, e.g. hitting the descriptor limit somehow?

So, perhaps there should be a way out of the loop, e.g. a time out, and/or running this loop in a separate thread?

What would your recommend?  :-)

Reply:
I would recommend a "if counter > 100" then break. The value does not matter. I assume this will work fine as is in 99.9 percent of the cases, but other people will look at the code and wonder.

Reply:
@anthonyfok bump; I will merge this but you need to put in a break in the loop so it can exit in those special cases.

Reply:
Hi!  Sorry, I had a very busy week (preparing and working at a conference), and then I got lazy.  :-D 
 Thank you for the reminder!  I'd better get on it and add the counter.  Will update this pull request within an hour.